app/services/survey_response_processor.rb

```diff
@@ -40,9 +59,44 @@ def ak_action
     @ak_action ||= ActionRepository.get(@params[:meta][:action_id])
   end
 
+  # AK's API has an inconsistent format when it comes to creating vs updating actions with custom fields.
+  # * On creation: To pass a custom field it must be passed as a regular field prepending the `action_*` prefix.
+  # example: { action_my_field: 'hello world' }
+  # * On update: To pass a custom field it must be passed as a json object on the `fields` key.
+  # example: { fields: { my_field: 'hello world' }
+  #
+  # We're also prepending the `survey_` prefix to all custom fields.
+
   def update_params
-    @params[:params].clone.tap do |p|
-      p[:page] = @ak_action[:page_ak_id]
+    params = @params[:params].clone
+    params[:page] = ak_action[:page_ak_id]
+
+    # Move all params prefixed with `action_` to
+    # the fields key and remove the prefix
+    action_params = {}
+    params.each do |key, val|
+      if key =~ /\Aaction_.*/
+        new_key = key.gsub(/\Aaction_/, '')
+        action_params["survey_#{new_key}"] = val
+      end
+    end
+    params[:fields] = action_params
+
+    params
+  end
+
+  # Replace `action_` prefix with `action_survey_` prefix
+  def create_params
+    params = @params.clone
+    params[:params] = params[:params].clone
```

Comment:
I'm cloning it because it's not a good practice to modify objects that you receive via parameters, unless the method is very explicit about doing that. 

Also, I'm sharing the `@params` between two different methods `create_params` and `update_params`, and they both need to do different modifications.